router/voteCallbacks.go

```diff
@@ -257,13 +305,136 @@ func (r *Router) getEnvelopeList(request routerRequest) {
 	if request.ListSize == 0 {
 		request.ListSize = MaxListSize
 	}
-	nullifiers := r.vocapp.State.EnvelopeList(request.ProcessID, request.From, request.ListSize, true)
+	envelopes, err := r.Scrutinizer.GetEnvelopes(request.ProcessID)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot get envelope list: (%s)", err))
+		return
+	}
+	listBytes, err := proto.Marshal(envelopes)
+	if err != nil {
+		r.sendError(request, fmt.Sprintf("cannot marshal vote envelope list: (%s)", err))
+		return
+	}
 	var response types.MetaResponse
-	strnull := []string{}
-	for _, n := range nullifiers {
-		strnull = append(strnull, fmt.Sprintf("%x", n))
+	response.Content = listBytes
+	request.Send(r.buildReply(request, &response))
+}
+
+func (r *Router) getValidatorList(request routerRequest) {
+	var response types.MetaResponse
+	validators, err := r.vocapp.State.Validators(true)
```

Comment:
Lets store the validators list in memory when the App is created and add the modificators on the scrutinizer event callbacks